### DISCUSSION TITLE
Bond constrains in geometry optimization

### DISCUSSION THREAD

**Discussion by icamps** [grimme-lab/xtb#1092] [Q&A] [answered]
Hi,

For the molecular dynamic calculations, it is possible to set up the parameter `shake`, that can constrain the atomic bonds.

I did a geometry optimization with xTB of a linear metal nanocluster (four Cd atoms in a row), and the result is that the cluster was break in half with two fragments of two Cd atoms each. I previously did (for other research) geometry optimization using DFT (with SIESTA and GAUSSIAN) and the cluster remain with all the atoms connected.

My question: is there a similar parameter for geometry optimization in order to force atoms to remain bonded?

Regards,

Camps

**Comment by marcelmbn** [comment, marked as answer]
You can use, e.g., distance constraints, as described here:
https://xtb-docs.readthedocs.io/en/latest/xcontrol.html#constraining-potentials

These should work for both MDs and geometry optimizations.
